eth/eventdatahandler/event_data_handler.go

```diff
@@ -105,30 +106,30 @@ func (edh *EventDataHandler) HandleBlockEventsStream(blockEventsCh <-chan eventb
 		logger = logger.With(fields.Count(len(tasks)))
 		logger.Info("executing tasks")
 
-		// TODO:
-		// 1) find and remove opposite tasks (start-stop, stop-start, liquidate-reactivate, reactivate-liquidate)
-		// 2) find superseding tasks and remove superseded ones (updateFee-updateFee)
-		for _, task := range tasks {
-			edh.logger.Debug("going to execute task") // TODO: add more task details
-			if err := task(); err != nil {
+		cleanTaskList := cleanTaskList(tasks)
+		for _, task := range cleanTaskList {
+			edh.logger.With(
+				zap.String("event_type", task.GetEventType()),
+			)
+			edh.logger.Debug("going to execute remote task")
+			if err := task.Execute(); err != nil {
 				// TODO: We log failed task until we discuss how we want to handle this case. We likely need to crash the node in this case.
-				edh.logger.Error("failed to execute task", zap.Error(err))
+				edh.logger.Error("failed to execute remote task")
```

Comment:
Let's use `logger` and keep the error logged unless there's a reason not to log it
```suggestion
				logger.Error("failed to execute remote task", zap.Error(err))
```